File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <map>
#include <algorithm>

using namespace std;

vector<int> solution(int n, vector<string> words) 
{
    vector<int> result(2); 
    
    vector<vector<string>> vec(n+1);
    vector<string> w;
    for(int i=0; i<words.size(); i++)
    {
        if(i >= 1 && words[i-1][words[i-1].size()-1] != words[i][0])
        {
            cout << 1;
            int tmp = (i+1) % n;
            if(tmp != 0)
            {
                vec[tmp].push_back(words[i]);
                result[0] = tmp;
                result[1] = vec[tmp].size(); 
            }
            else
            {
                vec[n].push_back(words[i]);
                result[0] = n;
                result[1] = vec[n].size(); 
            }
            break;
        } 
        else if(find(w.begin(), w.end(), words[i]) == w.end()) 
        {            
            int tmp = (i+1) % n;
            if(tmp != 0)
            {
                vec[tmp].push_back(words[i]);
            }
            else
            {
                vec[n].push_back(words[i]);
            }
        }
        else if(find(w.begin(), w.end(), words[i]) != w.end())
        {
            int tmp = (i+1) % n;
            if(tmp != 0)
            {
                vec[tmp].push_back(words[i]);
                result[0] = tmp;
                result[1] = vec[tmp].size(); 
            }
            else
            {
                vec[n].push_back(words[i]);
                result[0] = n;
                result[1] = vec[n].size(); 
            }
            break;
        }
        w.push_back(words[i]);
    }

    
    return result;
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.cs
using System;
using System.Collections.Generic;

class Solution
{
    public int[] solution(int n, string[] words)
    {
        List<string> wordList = new List<string>();
        int[] result = new int[2];
        for(int i=0; i<words.Length; i++)
        {    
            if(!wordList.Contains(words[i]))
            {
                wordList.Add(words[i]);
            }
            else
            {
                result[0] = i % n + 1;
                result[1] = i / n + 1;
                break;
            }
            
            if(i >= 1)
            {
                if(words[i][0] != words[i - 1][words[i - 1].Length - 1])
                {
                    result[0] = i % n + 1;
                    result[1] = i / n + 1;
                    break;
                }
            }
        }
        return result;
    }
}